### PR TITLE
SRCH-1038 update NoResultsWatcher for Elasticsearch 5.6

### PR DESCRIPTION
**Pull request by MothOnMars**
- update NoResultsWatcher for Elasticsearch 5.6
- upgrade Elasticsearch gems
- convert `groovy` scripts to `painless`

### CODE REVIEW

**Comment by MothOnMars** [comment]
Closing to investigate/resolve a issue found in testing.

**Comment by MothOnMars** [comment]
Re-opening. The issue I found was due to the test setup on demo boxes, which I'll have to resolve in a separate story.

File: spec/models/logstash_queries/watcher_no_results_query_spec.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+describe WatcherNoResultsQuery do
+  let(:query_args) do
+    {
+      affiliate_name: 'affiliate_name',
+      time_window: '1w',
+      min_doc_count: 50,
+      query_blocklist: 'foo, bar, baz biz',
```
Comment (peggles2) [inline spec/models/logstash_queries/watcher_no_results_query_spec.rb]: did you mean for baz biz to be 1? or is it baz, biz?

Comment (MothOnMars) [reply spec/models/logstash_queries/watcher_no_results_query_spec.rb]: Yeah, `baz biz` should be one element. The purpose is to test that the blocklist elements are being separated by commas into an array:
```ruby
"params.query.raw": [
                  "foo",
                  "bar",
                  "baz biz"
                ]
```

**Pull request merged**